user_modules/shared_testcases/testCasesForExternalUCS.lua

```diff
@@ -0,0 +1,179 @@
+        utils.createJsonFileFromTable(utils.pts, ptu_file_name)
+        EXPECT_HMINOTIFICATION("SDL.OnStatusUpdate", { status = "UPDATING" }, { status = status }):Times(2)
+        test["mobileSession"..id]:ExpectNotification("OnSystemRequest", { requestType = "PROPRIETARY" })
+        :Do(
+          function()
+	          local corIdSystemRequest = test["mobileSession"..id]:SendRPC("SystemRequest",
+	            { requestType = "PROPRIETARY", fileName = policy_file_name }, ptu_file_name)
+	          EXPECT_HMICALL("BasicCommunication.SystemRequest")
+	          :Do(
+	            function(_, d)
+	              test.hmiConnection:SendResponse(d.id, "BasicCommunication.SystemRequest", "SUCCESS", { })
+	              test.hmiConnection:SendNotification("SDL.OnReceivedPolicyUpdate",
+	                { policyfile = policy_file_path .. "/" .. policy_file_name })
+	            end)
+	            test["mobileSession"..id]:ExpectResponse(corIdSystemRequest, { success = true, resultCode = "SUCCESS"})
+          end)
+      end)
+    os.remove(ptu_file_name)
+  end
+
+	function utils.startSession(test, id)
+    test["mobileSession"..id] = mobile_session.MobileSession(test, test.mobileConnection)
+    test["mobileSession"..id]:StartService(7)
+  end
+
+	function utils.registerApp(test, id)
+		local RAIParams = config["application"..id].registerAppInterfaceParams
+    local corId = test["mobileSession"..id]:SendRPC("RegisterAppInterface", RAIParams)
+    EXPECT_HMINOTIFICATION("BasicCommunication.OnAppRegistered",
+      { application = { appName = RAIParams.appName } })
+    :Do(
+      function(_, d)
+        utils.HMIAppIds[RAIParams.appID] = d.params.application.appID
+      end)
+    test["mobileSession"..id]:ExpectResponse(corId, { success = true, resultCode = "SUCCESS" })
+    :Do(
+      function()
+        test["mobileSession"..id]:ExpectNotification("OnHMIStatus",
+          { hmiLevel = "NONE", audioStreamingState = "NOT_AUDIBLE", systemContext = "MAIN" })
+        test["mobileSession"..id]:ExpectNotification("OnPermissionsChange")
+      end)
+	end
+
+	function utils.activateApp(test, id)
+		local appId = config["application"..id].registerAppInterfaceParams.appID
+		local reqId = test.hmiConnection:SendRequest("SDL.ActivateApp", { appID = utils.HMIAppIds[appId] })
+    EXPECT_HMIRESPONSE(reqId)
+    :Do(
+      function(_, d1)
+        if d1.result.isSDLAllowed ~= true then
+          local reqId2 = test.hmiConnection:SendRequest("SDL.GetUserFriendlyMessage",
+            { language = "EN-US", messageCodes = { "DataConsent" } })
+          EXPECT_HMIRESPONSE(reqId2)
+          :Do(
+            function()
+              test.hmiConnection:SendNotification("SDL.OnAllowSDLFunctionality",
+                { allowed = true, source = "GUI", device = { id = config.deviceMAC, name = "127.0.0.1" } })
+              EXPECT_HMICALL("BasicCommunication.ActivateApp")
+              :Do(
```

Comment:
@dboltovskyi, in scripts usually function is listed in the same row as :Do(
:Do(function()